src/extensions/email_provider_rate_limit_extension.ts

```diff
@@ -0,0 +1,22 @@
+import { flattenErrors, statusCodeForError } from "lib/graphqlErrorHandler"
+
+// Log in `extensions` if this request was by an email provider and resulted
+// in a 429 rate limit error from a downstream request made using `fetch`.
+export const emailProviderRateLimitExtension = (_documentAST, result) => {
+  const extensions = {}
+
+  if (result.errors && result.errors.length) {
+    result.errors.some((err) => {
+      flattenErrors(err).some((e) => {
+        const httpStatusCode = statusCodeForError(e)
+        if (httpStatusCode === 429) {
+          extensions["emailProviderLimited"] = true
+          return true
+        }
+        return false
```

Comment:
Do these `true`/`false` return values do anything?

Reply:
it's just the early-returning of the `.some` iterator. I can double-check this snippet (it may or may not have been AI-generated).

Reply:
Can we use something other than `some` here? To joey's point, return true / false means that we should / should not return a result from some to be collected into a variable. If we're simply iterating over an array to check for values to be merged into an object, we can use `forEach`, which doesn't return anything and reads more correctly.

Reply:
Something like: 

```tsx
const extensions = result.errors?
  .flatMap(flattenErrors) 
  .reduce((acc, error) => {
    const httpStatusCode = statusCodeForError(error)
    
    if (httpStatusCode === 429) {
      acc.emailProviderLimited = true
    }
    
    return acc
  }, {})

return extensions
```